### PR TITLE
Implements multiple suggested clippy lints

### PR DESCRIPTION
**Pull request by ZeWaka**
These were the lints mostly:
* match_like_matches_macro
* mem_replace_with_default
* match_ref_pats
* single_char_push_str
* len_zero

There were also a few unnecessary return lints that popped up, but I'm not sure if you want to remove those.

### CODE REVIEW

File: src/dreammaker/ast.rs [hunk 1]
```diff
@@ -552,14 +552,13 @@ pub enum Term {
 
 impl Term {
     pub fn is_static(&self) -> bool {
-        return match self {
-            Term::Null |
-            Term::Int(_) |
-            Term::Float(_) |
+        matches!(self,
+            Term::Null      |
```
Comment (SpaceManiac) [inline src/dreammaker/ast.rs:556]: not a fan of the alignment being used here and on the other `matches!`, not sure if that was you or the tool though

**Comment by SpaceManiac** [comment]
> unnecessary return lints

probably fine to fix, at least some of them are leftovers from refactors

File: src/dreammaker/ast.rs (hunk 1 above)
Comment (ZeWaka) [reply src/dreammaker/ast.rs:556]: It was me, how would you prefer it to look like?

**Review by spookydonut — APPROVED** [review]

File: src/dreammaker/ast.rs (hunk 1 above)
Comment (SpaceManiac) [reply src/dreammaker/ast.rs:556]: `|` either at the end of the line without spaces, or at the beginning of the following line...

File: src/dreammaker/dmi.rs [hunk 2]
```diff
@@ -51,13 +51,10 @@ impl Dir {
...
     pub fn is_diagonal(self) -> bool {
-        match self {
-            Dir::North |
-            Dir::South |
-            Dir::East |
-            Dir::West => false,
-            _ => true
-        }
+        !matches!(self, Dir::North  |
+                        Dir::South  |
```
Comment (SpaceManiac) [inline src/dreammaker/dmi.rs:55]: and preferably a single indent here rather than hanging alignment

**Review by SpaceManiac — APPROVED** [review]

**Pull request merged**